src/shell/pipe_exec/command_not_found.rs

```diff
@@ -1,32 +1,24 @@
-use super::super::{Capture, Function, Shell};
-use parser::shell_expand::expand_string;
-use std::io::Read;
+use super::super::{Capture, Function, FunctionError, Shell};
 use std::process;
 use sys;
 
-pub(crate) fn command_not_found(shell: &mut Shell, command: &str) -> Option<String> {
-    let function = shell.functions.get("COMMAND_NOT_FOUND")? as *const Function;
+pub(crate) fn command_not_found(shell: &mut Shell, command: &str) -> bool {
+    let function = match shell.functions.get("COMMAND_NOT_FOUND") {
+        Some(func) => func as *const Function,
+        None => return false
+    };
 
-    let mut output = None;
-
-    match shell.fork(Capture::Stdout, |child| unsafe {
-        let _ = function.read().execute(child, &["ion", command]);
-    }) {
-        Ok(result) => {
-            let mut string = String::new();
-            match result.stdout.unwrap().read_to_string(&mut string) {
-                Ok(_) => output = Some(string),
-                Err(err) => {
-                    eprintln!("ion: error reading stdout of child: {}", err);
-                }
-            }
-        },
-        Err(err) => {
-            eprintln!("ion: fork error: {}", err);
+    if let Err(err) = shell.fork(Capture::None, |child| {
+        let result = unsafe { function.read() }.execute(child, &["ion", command]);
+        if let Err(FunctionError::InvalidArgumentCount) = result {
```

Comment:
May also want to handle `FunctionError::InvalidArgumentType(Primitive, String)`. The `FunctionError` type implements `Display`, so you should be able to just write this as:

```rust
if let Err(why) = result {
    eprintln!("ion: COMMAND_NOT_FOUND function hook: {}", why);
}

Reply:
Didn't realize ion had types. That's pretty cool!